instrumentation/couchbase/couchbase-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/couchbase/v2_0/CouchbaseRequest.java

```diff
@@ -7,18 +7,27 @@
 
 import com.google.auto.value.AutoValue;
 import io.opentelemetry.instrumentation.api.db.SqlStatementInfo;
-import java.lang.reflect.Method;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 import javax.annotation.Nullable;
 
 @AutoValue
 public abstract class CouchbaseRequest {
 
-  public static CouchbaseRequest create(@Nullable String bucket, Method method) {
-    Class<?> declaringClass = method.getDeclaringClass();
-    String className =
-        declaringClass.getSimpleName().replace("CouchbaseAsync", "").replace("DefaultAsync", "");
-    String operation = className + "." + method.getName();
-
+  private static final ClassValue<Map<String, String>> methodOperationNames =
```

Comment:
If possibly, can we compare performance vs just a normal weak keys cache with Method as the key? Seems like it could be faster or maybe not.

Reply:
Unfortunately our weak cache uses identity comparison for keys: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/3607#discussion_r676509051

The benchmark does not really look favorable too:

```
Benchmark                                                Mode  Cnt           Score            Error   Units
TestBenchmark.cache                                      avgt   10           0.081 ±          0.007   us/op
TestBenchmark.noCache                                    avgt   10           0.168 ±          0.020   us/op
TestBenchmark.noReflection                               avgt   10           0.109 ±          0.042   us/op
TestBenchmark.noReflectionCache                          avgt   10           0.013 ±          0.002   us/op

TestBenchmark.weakCache                                  avgt   10           1.058 ±          1.199   us/op
TestBenchmark.weakCache:heap.total.after                 avgt   10  3459252224.000 ± 1222610621.767   bytes
TestBenchmark.weakCache:heap.total.before                avgt   10  3164182937.600 ± 1424397272.793   bytes
TestBenchmark.weakCache:heap.used.after                  avgt   10  1629694261.600 ±  803347502.951   bytes
TestBenchmark.weakCache:heap.used.before                 avgt   10  1315928346.400 ±  807307188.320   bytes
TestBenchmark.weakCache:·gc.alloc.rate                   avgt   10         275.637 ±         77.225  MB/sec
TestBenchmark.weakCache:·gc.alloc.rate.norm              avgt   10         383.037 ±        148.914    B/op
TestBenchmark.weakCache:·gc.churn.G1_Eden_Space          avgt   10         221.034 ±        138.468  MB/sec
TestBenchmark.weakCache:·gc.churn.G1_Eden_Space.norm     avgt   10         270.356 ±        165.842    B/op
TestBenchmark.weakCache:·gc.count                        avgt   10          11.000                   counts
TestBenchmark.weakCache:·gc.time                         avgt   10        1453.000                       ms
```

Reply:
Oh those are really interesting comments over there:

> `Class.getDeclaredMethod` returns a different instance each time

🤯

> ByteBuddy does not cache the Method instance. When referencing @Advice.Origin Method method it effectively emits Foo.class.getMethod("name", ...) on each invocation

😬 (now I see why #3821)

Reply:
Neat thanks for clarifying!